### PR TITLE
daemon/logger/awslogs: add CloudWatch entity support

### LINKED ISSUE #52622: [cloudwatchlogs]awslogs driver does not support entity attributes/keyattributes
### Description

CloudWatch Logs allows you to search telemetry for related resources by adding an entity.

[PutLogEventsInput](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/cloudwatchlogs#PutLogEventsInput) includes an entity field, 
```go
type PutLogEventsInput struct {

	// The log events.
	//
	// This member is required.
	LogEvents []types.InputLogEvent

	// The name of the log group.
	//
	// This member is required.
	LogGroupName *string

	// The name of the log stream.
	//
	// This member is required.
	LogStreamName *string

	// The entity associated with the log events.
	Entity *types.Entity

	SequenceToken *string
	// contains filtered or unexported fields
}
```

but awslogs driver does not support entities yet.
https://github.com/moby/moby/blob/4c53ae5c2efd6881e8e513b20fb507b50a542890/daemon/logger/awslogs/cloudwatchlogs.go#L703-L708

Can you help with this?

-----------------------------------------

[cloudwatch-agent config]
https://github.com/aws/amazon-cloudwatch-agent/blob/99245a47610e4bacec2df50e1fb4b1c0303bb379/translator/translate/logs/logs.go#L39-L40

[cloudwatch-agent config]
https://github.com/aws/amazon-cloudwatch-agent/blob/99245a47610e4bacec2df50e1fb4b1c0303bb379/plugins/processors/awsentity/entityattributes/entityattributes_test.go#L25

### PR DESCRIPTION
**Pull request by sahilsGit**
closes #52622

**Added**

  - awslogs-entity-service-name: the entity's Name key attribute
  - awslogs-entity-environment:  the entity's Environment key attribute
  - awslogs-entity-attributes:   a comma-separated key=value list of
                                 non-identifying entity attributes

The entity type is fixed to "Service", matching how the CloudWatch agent
attributes log telemetry. Service name and environment support
Go-template placeholders as awslogs-stream (for example {{.Name}}). When no
entity option is set, no Entity is sent.

A rejected entity (RejectedEntityInfo in the response) is logged as a warning
rather than treated as an error, since the log events are still accepted.

**- What I did**

Added support for the CloudWatch `Entity` field to the `awslogs` logging
driver. The driver now lets users attach entity metadata to log events so that
CloudWatch can associate container logs with a service in the "Explore related"
feature. Previously the `Entity` field of `PutLogEvents` was never populated.

**- How I did it**

- `parseEntity` builds a `types.Entity` from the new log options and stores it
  on the `logStream`; it's attached to every `PutLogEvents` request.
- Service name and environment go through the existing template engine
  (`templates.NewParse`, as used by `awslogs-stream`), so `{{.Name}}`-style
  placeholders resolve per container.
- `awslogs-entity-attributes` is parsed as a `key=value` list, enforcing the
  CloudWatch limits (max 10; key/value lengths).
- `ValidateLogOpt` rejects a half-configured entity (service name and
  environment must be set together).
- `RejectedEntityInfo` in a response is logged as a warning — the log events
  are still accepted, so it isn't an error.
- No entity option set → no `Entity` sent; existing behavior unchanged.

**- How to verify it**
`go test -v -run Entity ./daemon/logger/awslogs/`

```markdown changelog
Add support for attaching service names, environments, and custom CloudWatch entity attributes to logs from the `awslogs` logging driver.
```

### CODE REVIEW

**Comment by sahilsGit** [comment]
**Decision Taken** 
The entity type is fixed to `Service` mirrors cloudWatch-agent, let me know your thoughts on this.

The amazon-cloudwatch-agent also sends `AwsAccountId` inside keyAttributes, but the docs doesn't talk about supporting anything like that, so I skipped here.
https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_Entity.html

**Comment by vvoland** [comment]
@arukiidou would you be able to test this?

File: daemon/logger/awslogs/cloudwatchlogs.go [hunk 1]
```diff
@@ -785,6 +928,15 @@ func ValidateLogOpt(cfg map[string]string) error {
 		}
 	}
 
+	serviceNameSet := cfg[entityServiceNameKey] != ""
+	environmentSet := cfg[entityEnvironmentKey] != ""
+	if serviceNameSet != environmentSet {
```
Comment (vvoland) [inline daemon/logger/awslogs/cloudwatchlogs.go]: What if both are false? This would also fail later, so we probably should also verify it here

Comment (sahilsGit) [reply daemon/logger/awslogs/cloudwatchlogs.go]: done

**Comment by arukiidou** [comment]
> @arukiidou would you be able to test this?

I want to this a test, 
but haven't our cloud environment is ready yet.

I've just started the design, probably will be ready to able test for a few more months.

**Comment by vvoland** [comment]
@sahilsGit @arukiidou  I added 3 commits handling some edge cases - let me know if they make sense! Otherwise feel free to drop them.

**Comment by sahilsGit** [comment]
> @sahilsGit @arukiidou I added 3 commits handling some edge cases - let me know if they make sense! Otherwise feel free to drop them.

New 3 commits are good, let's keep them

**Review by vvoland — APPROVED** [review]

**Comment by arukiidou** [comment]
@vvoland 
LGTM!

**Review by thaJeztah — APPROVED** [review]
LGTM, thx!
